numpy/compat/py3k.py

```diff
@@ -201,6 +198,10 @@ def npy_load_module(name, fn, info=None):
     def _PurePath__fspath__(self):
         return str(self)
 
+    # Pathlib isn't exactly a cheap import. Lazy import it here in case
+    # users are running python older than 3.6
+    from pathlib import PurePath
```

Comment:
May as well move this within the functions below. Looks like you can drop the `is None` checks now that 3.5 is our oldest supported version

Reply:
sometimes i am really blind. I jsut managed to install python 3.5. I'll try to run a few tests locally.

Reply:
This comment still applies - can you move this import inside `__subclasshook__` and `os_fspath` for even more performance?

Reply:
i mean sure, it just depends on how `unpythonic` you want to be. Striking a balance between style and speed really depends on who you talk to. Will anybody using less than python 3.7 be using numpy 1.18???

Reply:
At the point your import isn't at the top of the file, hiding it inside an `if` vs inside a `def` doesn't in my opinion rate very differently on a pythonicity scale - especially since you already import pathlib inside a function elsewhere in this PR.

Reply:
Right, but that import in a function comes attached with a warning, which anybody maintaining code will be addressing. So in the "Optimistic case" of "updated code using numpy" that import never gets hit for any supported version of python. 

But for python 3.6, the `import` statement in the `if` will always be hit. One might argue that letting a use hit the import statement in every call to `os_PahtLike` might __hurt__ the performance of 3.6. Adding warnings definitely hurts performance, but we are giving users a clear way out of them.

Reply:
My assumption has been that reimports are basically free, as they just hit the module cache. Am I mistaken?

Reply:
Your argument seems sound, probably better to leave this as you have it:
```python
In [89]: def with_import(p):
    ...:     from pathlib import Path
    ...:     return isisntance(p, Path)
    ...:

In [90]: from pathlib import Path

In [91]: def without_import(p):
    ...:     return isinstance(p, Path)
    ...:

In [94]: %timeit with_import(1)
2.52 µs ± 199 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [95]: %timeit without_import(1)
341 ns ± 27.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```